_episodes/02-benchmark-and-scaling.md

```diff
@@ -19,27 +19,244 @@ keypoints:
+As a first exercise, we will start with the simplest Lennard-Jones (LJ) system as provided in the 'bench' directory of the latest LAMMPS distribution. Following this, you will be given another example and you will follow the same set of exercises and compare your results with some 'known' benchmark.
+
+> ## Run a LAMMPS job in a HPC
+> Can you list bare minimum files that you need to schedule a LAMMPS job in a HPC?
+> > ## Solution
+> > For running a LAMMPS job, we must need an input file, empirical potential file (optional), and a data file (optional). We need these optional files only when the empirical potential model parameters and the molecular coordinates are not defined within the LAMMPS input file. In addition, for submitting a LAMMPS job to a HPC queue, we need a batch file. In general, each HPC uses a queueing system to manage all the jobs. Two such popular ones are PBS and Slurm. 
+> {: .solution}
+{: .challenge}
 
-> ## Benchmarking
+The input file for the LJ-system is given below: 
+```
+{% include /snippets/ep02/in.lj %}
+```
+{: .Input}
+
+
+The timing information for this run with both 1 and 4 processors is also provided with LAMMPS distribution. So, to benchmark it would be wise to run the same job with same processor settings. Let us now create a batch file to submit this job. 
+
+
+> ## Batch file to submit a LAMMPS job
+>
+> Here is shown the header of a SLURM script to submit a LAMMPS job using 2 nodes and 48 processors. Can you modify the necessary fields in this script to submit the job using 4 processors?
+>
+> ~~~
+> #!/bin/bash -x
+> {{ site.sched_comment }} --account=myAccount
+> {{ site.sched_comment }} --nodes=2
+> {{ site.sched_comment }} --ntasks-per-node=24
+> {{ site.sched_comment }} --output=mpi-out.%j
+> {{ site.sched_comment }} --error=mpi-err.%j
+> {{ site.sched_comment }} --time=00:05:00
+> {{ site.sched_comment }} --partition=myQueue
+>
+> ... ... ...
+> ... ... ...
+> ~~~
+> {: .input}
 > 
-> From what you have learned in this session, try a benchmark of `this`.
+> > ## Solution
+> > ~~~
+> > {{ site.sched_comment }} --nodes=1
+> > {{ site.sched_comment }} --ntasks-per-node=4
+> > ~~~
+> > {: .input}
+> {: .solution}
+{: .challenge}
+
+Okay, the rest part of the batch file should now be used to set the LAMMPS environment variables and finally we need to invoke the LAMMPS executable for the run. In all modern HPCs, package specific environment variables are loaded or unloaded using modules. Module is a piece of code that is load or unloaded into Kernel as required thus eliminating the requirement of system reboot each time you need to extend the functionalities of the Kernel. Here we show an example code on how to use the 'module' command to load LAMMPS in Jureca.
```

Comment:
Instead of 'example code on how to use the 'module' command to load LAMMPS in Jureca.', this would likely need changing to an abstraction, as different presenters would be using different machines.

config.yml option of `{{ site.host_name }}`